tests/headers/CMakeLists.txt

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2015 Hartmut Kaiser
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+# collect a list of all header files in the source tree
+file(GLOB_RECURSE headers "${PROJECT_SOURCE_DIR}/hpx/*hpp")
```

Comment:
We should probably exclude everything in directories in "detail" directories, those are not meant to be included by anyone knowing exactly what they do anyways.